sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -2832,16 +2832,21 @@ SIREPO.app.directive('jobsList', function(requestSender, appState, $location, $s
             }
 
             function getRow(row, nameIndex, simulationIdIndex, appIndex) {
+                function formatString(str) {
+                    if (! str) {
+                        return 'n/a';
+                    }
+                    return str;
+                }
+
+                var typeDispatch = {
+                    DateTime: appState.formatDate,
+                    Time: appState.formatTime,
+                    String: formatString,
+                };
                 var h = '';
                 for (var i = getStartIndex(); i < row.length; i++) {
-                    var v = row[i];
-                    var t = $scope.data.header[i][1];
-                    if (t === 'DateTime') {
-                        v = appState.formatDate(v);
-                    }
-                    else if (t === 'Time') {
-                        v = appState.formatTime(v);
-                    }
+                    var v = typeDispatch[$scope.data.header[i][1]](row[i]);
```

Comment:
Is there a bug here if the string could contain html?

Would be nice to encapsulate the n/a, since it is used in two different files. The only odd case is formatTime '0'. If we tested the result of the formatter for falsey, it might work. Then, we'd just have to return null in the case of falsey in formatDate. formatTime could stay the same, and formatString could either return null for falsey or just falsey.

Reply:
> Is there a bug here if the string could contain html?
What kind of bug are you thinking of? An injection attack? The server could send back html and we would render it (it doesn't do this currently).

https://github.com/radiasoft/sirepo/pull/2730/commits/0c277ac46a3f83391c6f896c3a5524cd57098632